File: Data/Model/TweetDay.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Birder.Data.Model
{
    public class TweetDay
    {
        [Key]
        public int TweetDayId { get; set; }

        public DateTime DisplayDay { get; set; }

        public DateTime CreationDate { get; set; }

        public DateTime LastUpdateDate { get; set; }

        public int BirdId { get; set; }

        public Bird Bird { get; set; }
    }
}

File: Birder/ClientApp/src/app/_models/TweetDay.ts
import { BirdSummaryViewModel } from './BirdSummaryViewModel';

export interface TweetDay {
    tweetDayId: number;
    songUrl: string;
    displayDay: Date | string;
    creationDate: Date | string;
    lastUpdateDate: Date | string;
    bird: BirdSummaryViewModel;
}

export interface TweetArchiveDto {
    totalItems: number;
    items: TweetDay[];
}
